Modules/Generators/ProjectGenerator/Sources/ProjectGenerator/ComponentPackageProvider.swift

```diff
@@ -170,11 +170,11 @@ public struct ComponentPackageProvider: ComponentPackageProviderProtocol {
         
         return .init(package: .init(name: packageName,
                                     defaultLocalization: defaultLocalization,
-                                    iOSVersion: component.iOSVersion,
-                                    macCatalystVersion: component.macCatalystVersion,
-                                    macOSVersion: component.macOSVersion,
-                                    tvOSVersion: component.tvOSVersion,
-                                    watchOSVersion: component.watchOSVersion,
+                                    iOSVersion: component.platforms.iOSVersion,
```

Comment:
```suggestion
                                    iOSVersion: component.platforms.iOSVersion ?? projectConfiguration.platforms.iOSVersion,
```

This change for all those lines should be enough to make this work. You can revert injecting the document in the ProjectConfigurationView after doing this.

Reply:
Can you check [57ff8b8](https://github.com/Tawa/Phoenix/pull/74/commits/57ff8b8a9258fca074db8172f68ae99a2814d1c6) and [5f8e9b0](https://github.com/Tawa/Phoenix/pull/74/commits/5f8e9b08e4faa05dd5cf591617b71b45389e4dd5)?